### PR TITLE
[now-next] Add testing for export handling

### PR DESCRIPTION
**Pull request by ijjk**
Follow up to #3431 adding tests for this behavior

Tests to add:
- [x] custom routes: redirects
- [x] custom routes: rewrites
- [x] confirming each test is actually a `next export` deploy somehow

### CODE REVIEW

File: packages/now-next/src/index.ts [hunk 1]
```diff
@@ -355,10 +355,11 @@ export const build = async ({
   }
 
   const exportIntent = await getExportIntent(entryPath);
-  if (exportIntent) {
-    const { trailingSlash } = exportIntent;
+  const userExport = await getExportStatus(entryPath);
+
+  if (exportIntent || userExport) {
+    const { trailingSlash = true } = exportIntent || {};
```
Comment (Timer) [inline packages/now-next/src/index.ts]: Trailing slash now defaults to false in Next 9+.

**Review by Timer — CHANGES_REQUESTED** [review]

File: packages/now-next/src/index.ts (hunk 1 above)
Comment (ijjk) [reply packages/now-next/src/index.ts]: Updated

File: packages/now-next/src/index.ts [hunk 2]
```diff
@@ -378,18 +379,12 @@ export const build = async ({
     }
 
     const resultingExport = await getExportStatus(entryPath);
-    if (!resultingExport) {
+    if (!resultingExport || resultingExport.success !== true) {
```
Comment (Timer) [inline packages/now-next/src/index.ts]: These errors were intentionally separate.

If the file is missing, there's a Next.js bug (truthy check) -- this should be considered an invariant.

If the file exisits and success is false, their pages have a bug.

**Review by Timer — CHANGES_REQUESTED** [review]

File: packages/now-next/src/index.ts (hunk 2 above)
Comment (ijjk) [reply packages/now-next/src/index.ts]: Broke them back apart

**Review by Timer — APPROVED** [review]
LGTM once tests are passing.

**Pull request merged**